core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -181,14 +181,36 @@ private[spark] class CoarseGrainedExecutorBackend(
 
 private[spark] object CoarseGrainedExecutorBackend extends Logging {
 
-  private def run(
+  case class CoarseGrainedExecutorBackendArguments(
       driverUrl: String,
       executorId: String,
       hostname: String,
       cores: Int,
       appId: String,
       workerUrl: Option[String],
-      userClassPath: Seq[URL]) {
+      userClassPath: mutable.ListBuffer[URL])
+
+  def main(args: Array[String]): Unit = {
+    val arguments = parseArguments(args, this.getClass.getCanonicalName)
+
+    val createFn: (RpcEnv, SparkEnv) => CoarseGrainedExecutorBackend = { case (rpcEnv, env) =>
+      new CoarseGrainedExecutorBackend(rpcEnv, arguments.driverUrl, arguments.executorId,
+        arguments.hostname, arguments.cores, arguments.userClassPath, env)
+    }
+    run(arguments.driverUrl, arguments.executorId, arguments.hostname, arguments.cores,
+      arguments.appId, arguments.workerUrl, arguments.userClassPath, createFn)
+    System.exit(0)
+  }
+
+  def run(
+      driverUrl: String,
+      executorId: String,
+      hostname: String,
+      cores: Int,
+      appId: String,
+      workerUrl: Option[String],
+      userClassPath: Seq[URL],
```

Comment:
This is actually not used now. I'm almost suggesting you should just have `CoarseGrainedExecutorBackendArguments` as an argument here...

Reply:
Ah yes nice suggestion. I missed it. Will address.